app/controllers/users/progress_controller.rb

```diff
@@ -0,0 +1,11 @@
+module Users
+  class ProgressController < ApplicationController
+    before_action :authenticate_user!
+
+    def destroy
+      user_id = params[:id]
+      ResetProgressJob.perform_async(user_id)
+      redirect_to edit_user_registration_path, notice: 'Your progress has been reset.'
```

Comment:
Great question on the copy here. Sorry, I overlooked the immediacy of when this would be ran when writing the story.

I don't think using future tense would be something we want to do without someway of telling them when it would be done. This is something I wouldn't mind us exploring in the future with webhook updates for when background jobs have completed. But I think its a bit too much for to do right now.

That leaves two options imo. 

1. We don't use a background job and just do it in the controller. It should be fine since we're using delete_all instead of deleting each lesson completion with destroy. 

2. We implement a queue priority as [described here](https://github.com/mperham/sidekiq/wiki/Advanced-Options) and set this job as the highest priority so its done almost immediately. We don't have many jobs at the moment so I think this would tide us over until we integrate webhook into the system.

Reply:
I hope that I am not making a mountain out of molehill here, as I do not think there is much of a delay. However, do you have a preference for which option to implement?

Also, I am currently using `destroy_all` in the ResetProgressJob, but can change it to `delete_all` if you want me to put it this in the controller (or if you want to be using `delete_all` in the background job).

Reply:
Whoops, sorry, I seen the `*_all` and thought it was the delete version 😆 

After thinking about it a bit more, I think we should probably avoid using the background job for now as we can't be certain it will always be immediate. It's unlikely we will have any associations of the lesson completions any time soon so I think it's safe to use delete_all which should make it immediate and not hold up the request.

Sorry for leading you down the wrong direction with this for suggesting a background job initially.

Reply:
@KevinMulhern No big deal. I'm glad I got experience creating and testing a background job. 

Would it make sense to create a service object instead of putting all of this logic in the controller? As I started making these changes, I'm realizing that I have limited experience testing controllers because I tend to put all the heavy logic in a service object (or something similar).

Reply:
👍  for putting it in a service, something similar to the admin services where we run the steps in a transaction should be perfect for this.